### PR TITLE
Add Converters

### PR DESCRIPTION
**Pull request by eugenebokhan**

### CODE REVIEW

File: Sources/Converters.swift [hunk 1]
```diff
@@ -316,24 +320,21 @@ class EluConverter: NodeConverter {
...
         else { throw ONNXGraph.Errors.noSuchOutput }
 
-        if #available(iOS 11.3, macOS 10.13.4, *) {
-            let sigmoid = MPSCNNNeuronExponentialNode(source: input)
-
-            graph.addFilter(sigmoid, outputShape: inputShape, withOutputs: node.output)
-        } else {
-            throw ONNXGraph.Errors.unknownNodeOpType(opType: "Exp")
-        }
+        let sigmoid = MPSCNNNeuronExponentialNode(source: input)
```
Comment (s1ddok) [inline Sources/Converters.swift]: sigmoid = exp?)

File: Sources/Converters.swift [hunk 2]
```diff
@@ -347,7 +348,24 @@ class AddConverter: NodeConverter {
...
 
-class MulConverter: NodeConverter {
+@available(iOS 11.0, macOS 10.13.0, tvOS 11.0, *)
+final class SubConverter: NodeConverter {
+    func convert(in graph: ONNXGraph, node: Onnx_NodeProto) throws {
+        guard
+            let input1 = graph.output(name: node.input[0]),
+            let input2 = graph.output(name: node.input[1]),
+            let inputShape = graph.shape(output: node.input[0]),
+            node.input.count == 2
```
Comment (Legonaftik) [inline Sources/Converters.swift:358]: Shouldn't we check `.count == 2` before accessing the elements of `node.input` by index?

File: Sources/Converters.swift [hunk 3]
```diff
@@ -361,7 +379,24 @@ class MulConverter: NodeConverter {
...
 
-class SigmoidConverter: NodeConverter {
+@available(iOS 11.0, macOS 10.13.0, tvOS 11.0, *)
+final class DivConverter: NodeConverter {
+    func convert(in graph: ONNXGraph, node: Onnx_NodeProto) throws {
+        guard
+            let input1 = graph.output(name: node.input[0]),
+            let input2 = graph.output(name: node.input[1]),
+            let inputShape = graph.shape(output: node.input[0]),
+            node.input.count == 2
```
Comment (Legonaftik) [inline Sources/Converters.swift:389]: Same

> Shouldn't we check .count == 2 before accessing the elements of node.input by index?

File: Sources/Converters.swift [hunk 4]
```diff
@@ -753,3 +813,101 @@ class DropoutConverter: NodeConverter {
...
+        graph.addFilter(abs, outputShape: inputShape, withOutputs: node.output)
+    }
+}
+
+@available(iOS 11.0, macOS 10.13.0, tvOS 11.0, *)
+final class HardSigmoidConverter: NodeConverter {
+    func convert(in graph: ONNXGraph, node: Onnx_NodeProto) throws {
+        guard
+            let input = graph.output(name: node.input[0]),
+            let inputShape = graph.shape(output: node.input[0])
```
Comment (Legonaftik) [inline Sources/Converters.swift:836]: Is it safe to access the elements by index here without checking `.count` like you previously did?
Maybe should add `guard let nodeInputFirst = node.input.first, ...`?

File: Sources/Converters.swift [hunk 5]
```diff
@@ -18,7 +18,8 @@ enum ConvWeightArray {
 
 // MARK: Convolutions
 
-@objc private class ConvDataSource: NSObject, MPSCNNConvolutionDataSource {
+@available(iOS 11.0, macOS 10.13.0, tvOS 11.0, *)
```
Comment (Legonaftik) [inline Sources/Converters.swift:21]: From the current podspec:
```
  s.ios.deployment_target = '11.0'
  s.osx.deployment_target = '10.13'
```
Looks like this and some other `@available` checks are redundant

**Comment by s1ddok** [comment]
@Legonaftik thanks for getting involved, this PR is on hold since we don't really use it right now. I will make sure we will take a look at your comments a bit later

**Pull request closed without merging**